### DISCUSSION TITLE
Should we implement ping/pong on both client and server

### DISCUSSION THREAD

**Discussion by cgvalayev** [enisdenjo/graphql-ws#464] [Q&A] [answered]
Originally we had ping pong initiated by the server in which case connection was killed by our load balancer which had 30 seconds keep alive.
We then added ping pong from the client and this helped to keep connection stable.

Whats the best practice here?

Also, any suggestions on keep alive config settings for the load balancers?

**Comment by enisdenjo** [comment, marked as answer]
Well, if the ping from client does the job - I'd recommend that. There are no conventions here, do what keeps the connection alive the best.

**Comment by cgvalayev** [comment]
Any suggestions for keep alive configuration? Should we make it a lot longer?

**Reply by enisdenjo** [reply]
No suggestions really, whatever works for you. Every load balancer/gateway/server/firewall is different.

**Comment by cgvalayev** [comment]
👍 thank you, we decided to keep aggressive client ping pong and less on the server. We will experiment with longer keep alive in the future, for now collecting data on connection and usage.
